pandas/tests/reshape/test_pivot.py

```diff
@@ -1991,12 +1993,13 @@ def test_pivot_string_func_vs_func(self, f, f_numpy):
     @pytest.mark.slow
     def test_pivot_number_of_levels_larger_than_int32(self):
         # GH 20601
+        # GH 26314: Change ValueError to PerformanceWarning
         df = DataFrame(
             {"ind1": np.arange(2 ** 16), "ind2": np.arange(2 ** 16), "count": 0}
         )
 
-        msg = "Unstacked DataFrame is too big, causing int32 overflow"
-        with pytest.raises(ValueError, match=msg):
+        msg = "The following operation may generate"
+        with tm.assert_produces_warning(PerformanceWarning, match=msg):
```

Comment:
does this test actually run?

Reply:
Yup. These are running on the slow builds.

```
6.56s call     pandas/tests/frame/test_stack_unstack.py::TestStackUnstackMultiLevel::test_unstack_number_of_levels_larger_than_int32
6.50s call     pandas/tests/reshape/test_pivot.py::TestPivotTable::test_pivot_number_of_levels_larger_than_int32

```